internal/controller/server_controller.go

```diff
@@ -120,6 +120,14 @@ func (r *ServerReconciler) reconcileExists(ctx context.Context, log logr.Logger,
 }
 
 func (r *ServerReconciler) delete(ctx context.Context, log logr.Logger, server *metalv1alpha1.Server) (ctrl.Result, error) {
+	if !controllerutil.ContainsFinalizer(server, ServerFinalizer) {
+		return ctrl.Result{}, nil
+	}
+
+	if server.Status.State == metalv1alpha1.ServerStateMaintenance {
+		log.V(1).Info("postponing delete as server is in Maintenance state")
```

Comment:
Skipping instead of postponing. We typically start the `Info()` logs with an upper case.